code/parse/sexp.cpp

```diff
@@ -2718,11 +2731,10 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 						break;
 
 					// look for mission
-					for (i = 0; i < Campaign.num_missions; i++)
-						if (!stricmp(CTEXT(z), Campaign.missions[i].name))
-							break;
+					count = count_named_items(CTEXT(z), Campaign.num_missions, Campaign.missions);
 
-					if (i >= Campaign.num_missions) {
+					// only check for a missing mission -- it's ok if the same mission appears multiple times in the campaign
```

Comment:
Thanks for including this comment. I was initially confused until I read it. :+1: